### PR TITLE
astore: Implement token URL parameter validation

### PR DESCRIPTION
**Pull request by minor-fixes**
This change implements token-based validation, as described in the [CX
Portal Artifact Connector API design
doc](https://docs.google.com/document/d/1Apvbt3LNpoLdUFtRliKZ8Ety-nXYCGNjfgSU2mde1o0/edit?tab=t.0#bookmark=id.7i6q2xwekj8q).

This involves:

* adding a flag `--token-public-key-file` that contains a public key
  that can be used to validate JWTs
* reading and parsing this file on startup and storing the resulting
  public key object inside `Server.options`
* adding a new route `/gt/`, which is similar to the `/g/` download
  route, except:
  * it is not protected by Oauth authentication
  * it enforces token-based authentication based on a URL parameter
* unit tests are added for the token validation code, to ensure that the
  proper token checks are in-place before allowing a download

Tested:
* added unit tests
* TODO: deploy to staging and manually test

Jira: REL-178

### CODE REVIEW

**Review by mhallmark — COMMENTED** [review]
I don't see where it validates the path or uid of the token to the requested resource. Is this intended to be put in later, or am I missing it?

**Comment by minor-fixes** [comment]
> I don't see where it validates the path or uid of the token to the requested resource. Is this intended to be put in later, or am I missing it?

...yeah fair, the way the JWT library works hides this somewhat. Take a look at https://github.com/enfabrica/enkit/pull/1212/files#diff-b3e47ec169762a4bda001e89708ae02e30749f1cf978b74378a8e100faaab490 - there's a custom `claims` struct defined which adheres to the [`ClaimsValidator`](https://pkg.go.dev/github.com/golang-jwt/jwt/v5#ClaimsValidator) interface. The UID is asserted in the custom `Validate` method, and the other items of the claim are "well-known" and validated by the JWT library by default. The validation happens at the call to `ValidateWithClaims` on the parser object created via the JWT library.

The unit test cases added run through a bunch of the edge cases, so hopefully it's clear that the validation is working as expected, even if it's a bit hidden?

**Review by mhallmark — APPROVED** [review]

File: astore/server/astore/retrieve.go [hunk 1]
```diff
@@ -14,35 +14,67 @@ import (
...
+	AuthTypeNone authType = iota
+	// Authentication is handled by credentials cookie + SSO
+	AuthTypeOauth
+	// Authentication is handled by JWT token parameter
+	AuthTypeToken
 )
 
-// DownalodArtifact turns an http.Request into an astore.RetrieveRequest, executes it, and invokes the specified handler with the result.
-func (s *Server) DownloadArtifact(prefix string, ehandler DownloadHandler, w http.ResponseWriter, r *http.Request) {
+// DownalodArtifact turns an http.Request into an astore.RetrieveRequest,
```
[the remark this answers is not in the document]
Comment (minor-fixes) [reply astore/server/astore/retrieve.go]: Done

File: astore/server/astore/factory.go [hunk 2]
```diff
@@ -98,6 +101,21 @@ func WithCredentialsJSON(json []byte) Modifier {
 	}
 }
 
+func WithTokenPublicKeyFile(path string) Modifier {
```
Comment (kjw-enf) [inline astore/server/astore/factory.go:104]: This doesn't appear used anywhere? no action required if it's going to be used shortly. just an observation
